### PR TITLE
Adding us_state column to models

### PR DESCRIPTION
**Pull request by jordan-springer**
# Description

tl;dr -- extracted the "state" value from `users_pii.properties` to model in Hydrone, now available if needed in `dim_users`

see validation numbers below for confirmation

## Links

Jira ticket(s): [dataops-916](https://codedotorg.atlassian.net/browse/DATAOPS-619)

## Testing story

- [x] Does your change include appropriate tests on key columns?
      eg.
      - `not_null`
      - `unique`
      - `dbt_utils.unique_combination_of_columns: , ["value","value","value"...]

**Note: when submitting a new model for review please make sure the following have been tested:**

1. The model compiles (`dbt build -m 'your_model'`)
         or: _has the dbt Cloud job succeeded?_
3. The model runs (`dbt run -m 'your_model'`)
4. The model produces accessible data in the DW (`select 1 from 'your_model'`)

## Privacy

- [x] 1.	Does this change involve the collection, use, or sharing of new Personal Data?
- [x] 2.    Do these data exist in the appropriate schema(s)? 
- [x] 3.	Does this change involve a new or changed use or sharing of existing Personal Data?
- [ ] 4.    Consider: will this data be visible on Tableau? will this data be surfaced in a report exported from Trevor?
- [ ] 5.    If yes to any of the above, please list the models, columns, and justification below:
      i.
      ii.
      iii. 

## PR Checklist:
**Note: if these are not all checked, the PR will be sent back.**

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code adheres to [style guide](https://docs.getdbt.com/best-practices/how-we-style/0-how-we-style-our-dbt-projects)👀 and is **[DRY](https://docs.getdbt.com/terms/dry)**
- [x] Code is well-commented (**please do not leave extraneous commentary in model code, if it is for the purpose of documentation, please relocate accordingly)
- [x] Appropriate documentation has been provided (see `.yml.`, did `dbt docs generate` succeed?)
- [x] New features are translatable or updates will not break up/downstream models
- [x] Relevant documentation has been added or updated (i.e. `dbt docs` has been updated successfully on [Github Pages](code-dot-org.github.io/analytics/)
- [x] Pull Request is labeled appropriately (eg. `chore/`, `feature/`, `fix/`)
- [x] Follow-up work items (including potential tech debt) are tracked and linked (if applicable)

### CODE REVIEW

File: dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@ with
 source as (
     select * 
     from {{ source('dashboard_pii', 'users') }}
-    where user_type is not null 
+    -- where user_type is not null
```
Comment (jordan-springer) [inline dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql:5]:
```suggestion
    where user_type is not null 
```

**Review by jordan-springer — COMMENTED** [review]
lgtm

File: dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql [hunk 2]
```diff
@@ -18,9 +18,12 @@ renamed as (
...
         updated_at,
         purged_at,
-        deleted_at
-    from source
-)
+        deleted_at,
+        json_extract_path_text(
+            lower(properties), 
+            'us_state',    
+             true)                  as us_state
```
Comment (bakerfranke) [inline dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql:25]: We've never used this state extraction from user properties to associate a state with a user before.  Are we sure we know the source of it?

In the past we've just used the state assigned by user_geos - it carries state (and sometimes city) if in the U.S.

I think you should double-check to see if this `us_state` aligns with the `user_geo.state` and/or figure out the source of this one.

File: dbt/models/marts/users/dim_users.sql [hunk 3]
```diff
@@ -21,15 +21,15 @@ final as (
         users_pii.races,
         users_pii.race_group,
         users_pii.gender_group,
-        ug.is_international,
+        users_pii.us_state,
```
Comment (bakerfranke) [inline dbt/models/marts/users/dim_users.sql:24]: why not just use `ug.state` ?

**Review by bakerfranke — CHANGES_REQUESTED** [review]

File: dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql (hunk 2 above)
Comment (jordan-springer) [reply dbt/models/staging/dashboard_pii/base/base_dashboard_pii__users.sql:25]: see ticket here: https://codedotorg.atlassian.net/browse/DATAOPS-668

File: dbt/models/marts/users/dim_users.sql (hunk 3 above)
Comment (jordan-springer) [reply dbt/models/marts/users/dim_users.sql:24]: this is user-entered, it is needed for CPA. `ug.state` is based on IP

[the remark this answers is not in the document]
Comment (jordan-springer) [reply dbt/models/marts/users/dim_users.sql:24]: this seems unnecessary, the field is stored as `user_state`, we can describe its nature in the docs

**Comment by jordan-springer** [comment]
@nataliazm99 what about states outside of the 50 US states?

**Comment by jordan-springer** [comment]
!data looks good(
image
)
